fall-2020/navigation.py

```diff
@@ -87,7 +88,86 @@ def navigate(self):
         Returns: True if actuation requests were successful, False if not
         Note: It may be good to notify the user if something unexpected happens!
         """
-        pass
+        # Getting current position of the TBM.
+        self.update_current_position()
+
+        if self.current_position == self.desired_position:
+            print("Current position is already at desired position.")
+            return True
+
+        # Looping through and comparing each corresponding coordinate in the two position tuples.
+        for coordinate_index in range(3):
+            # If x => check if TBM is behind, if y => check if TBM is left, if z => check if TBM is below.
+            if self.current_position[coordinate_index] < self.desired_position[coordinate_index]:
+                # Distance between the two corresponding x, y, or z values.
+                steer_distance = self.desired_position[coordinate_index] - self.current_position[coordinate_index]
+
+                # x value.
+                if coordinate_index == 0:
+                    if self.steering.move_forward():
+                        print("Actuation successful.")
+
+                    else:
+                        print("Forward actuation unsuccessful.")
+
+                # y value.
+                elif coordinate_index == 1:
+                    if self.steering.move_right(steer_distance):
+                        print("Actuation successful.")
+
```

Comment:
Don't think you need to put a blank line between the if and else statement blocks - also applies to all other if else's you have in this submission. It is a little more justified for the outer level of if/else, however for the nested ones I think it would be cleaner to not have a blank line.